File: StringsAndArrays/ImplementStrStr.py
# Implement strStr().

# Return the index of the first occurrence of needle in haystack, or -1 if needle is not part of haystack.

# My first approach
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if (haystack == []):
            return -1
        if (needle in haystack):
            k = needle[0]
            return haystack.index(k)
        else:
            return 0

# failed some test cases

# correct approach
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if needle in haystack:
            return haystack.index(needle)
        return -1

File: StringsAndArrays/ImplementStrStr.java
// Given two strings needle and haystack, return the index of the first occurrence of needle in haystack, or -1 if needle is not part of haystack.

class Solution {
    public int strStr(String haystack, String needle) {
        int m = needle.length();
        for (int i = 0; i <=haystack.length()-m; i++){
            if(haystack.substring(i, i+m).equals(needle)){
                return i;
            }
        }
        return -1;
    }
}

  
